openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -775,13 +799,397 @@ function scale(IndicatorObj) {
+                'paint': {
+                    'fill-color': colorsPal[i],
+                    'fill-opacity': 0.8,
+                    'fill-outline-color': '#000066'
+                },
+                'filter': ['all',['>', selectedIndicator, breaks[i]], ['<=', selectedIndicator, breaks[i+1]]]
+            });
+            // Create legend elements
+            $('#legendLables').append('<li><span style="background:'+colorsPal[i]+';"></span>'+breaks[i].toFixed(2)+'</li>');
+        }
+    //}
+
+    $('#absoluteSpinner').hide();
+
+    // Map interactivity
+    $('#map-tools').append(
+        '<div id="mapInfo"></div>'
+    );
+
+    map.on('click', function(e) {
+        map.featuresAt(e.point, { radius : 6}, function(err, features) {
+            if (err) throw err;
+            $('#mapInfo').empty();
+            for(var k in features[0].properties) {
+                $('#mapInfo').append(k+': '+features[0].properties[k]+'</br>');
+            }
+        });
+    });
+    colorPicker();
+}
+
+function setupLeafletMap() {
+    map = new L.Map('map', {
+        minZoom: 2,
+        scrollWheelZoom: false,
+        attributionControl: false,
+        maxBounds: new L.LatLngBounds(new L.LatLng(-90, -180), new L.LatLng(90, 180)),
+    });
+    map.setView(new L.LatLng(10, -10), 2).addLayer(baseMapUrl);
+}
+
+
+function leafletThematicMap(layerAttributes, allSVIThemes, allPrimaryIndicators, allRiskIndicators, weightChange) {
+
+    // Add main indicators to selection menu
+    $('#leafletThematicSelection').empty();
+    $('#leafletThematicSelection').append(
+        '<optgroup label="Indicators">'+
+        '<option>IRI</option>'+
+        '<option>SVI</option>'+
+        '<option>IR</option>'
+    );
+
+    // Add SVI themes to selection menu
+    $('#leafletThematicSelection').append('<optgroup label="SVI Indicators">');
+    for (var id = 0; id < allSVIThemes.length; id++) {
+        $('#leafletThematicSelection').append('<option>'+allSVIThemes[id]+'</option>');
+    }
+
+    try {
+        if (riskIndicators !== undefined) {
+            // Add IR themes to selection menu
+            $('#leafletThematicSelection').append('<optgroup label="IR Indicators">');
+            for (var ie = 0; ie < allRiskIndicators.length; ie++) {
+                $('#leafletThematicSelection').append('<option>'+allRiskIndicators[ie]+'</option>');
+            }
+        }
+    } catch (e) {
+        // continue
+    }
+
+    // Add all primary indicators to selection menu
+    $('#leafletThematicSelection').append('<optgroup label="Composite Indicators">');
+    for (var ic = 0; ic < allPrimaryIndicators.length; ic++) {
+        $('#leafletThematicSelection').append('<option>'+allPrimaryIndicators[ic]+'</option>');
+    }
+
+    // set the map selection menu to IRI or previously selected indicator value
+    if (selectedIndicator == undefined) {
+        // TODO fix this to use optgroup:first
+        //$('#leafletThematicSelection').val('IRI');
```

Comment:
there is already a ticket for this known issue: https://github.com/gem/oq-platform/issues/505